src/main/java/com/gmail/inverseconduit/bot/InteractionBot.java

```diff
@@ -26,13 +26,12 @@ public InteractionBot(ChatInterface chatInterface) {
 
     @Override
     public void start() {
-        executor.scheduleAtFixedRate(this::processInteractions, 200, 700, TimeUnit.MILLISECONDS);
+        // asynchronously enqueue the processing by blocking supplier
+        processingThread.submit(this::processInteractions);
```

Comment:
Maybe an appropriate rename can help here. I considered taking things a step further and using the same pattern as for shutdown by providing a default implementation for `start()` and delegate the processing to an abstract method.